### DISCUSSION TITLE
[C128] Why does a 'struct' always ends up in the standard BSS and not in my custom BSS?

### DISCUSSION THREAD

**Discussion by RetrofanC** [cc65/cc65#1928] [Q&A] [answered]
Hi, 

I find some curious behavior which I cannot explain for find in the documentation or on the internet.
It seems when I am using a 'struct' I cannot get that into a custom bss segment. 
Typically I use the #pragma in my C code for testing and when it works make it in native .asm 

 Now I made a new custom bss segment in the c128.cfg file and named it "COMB" 
Then I used it in my C prog around my struct definition.
This failed because no matter what I try it always goes into the standard BSS instead.

As a final test I copied some definition of the struct outside of the struct (bc, de, hl) and placed an x before them to make them unique.
Now if you look to the snipped of the C code and the generated .asm code you can see that the individual definitions are nice in my custom COMB. But the struct went back into the standard BSS..... 

**Question #1:** Is this correct/designed behavior that a struct always goes into the BSS segment ? 
**Question #2:** Is there any clever 'trick' I can use to still have the struct go into my custom defined COMB segment? 

```
#pragma bss-name (push, "COMB")
// Just taken out of the struct and added a x before the name to test COMBSS
	unsigned int xbc;     
	unsigned int xde;     
	unsigned int xhl;      

// The original struct which doesn't want to go into my custom COMBSS
struct ddregs
{
	unsigned int bc;       
	unsigned int de;       
	unsigned int hl;        
};


xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
Snipped from the .asm generated file 
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
0001E6r 1  52 42 00     	.byte	$52,$42,$00
0001E9r 1               L01CC	:=	L01E3+0

0001E9r 1               
0001E9r 1               **.segment	"BSS"**
000000r 1               
000000r 1               **.segment	"COMB"**
000000r 1               _xbc:
000000r 1  00 00        	.res	2,$00
000002r 1               _xde:
000002r 1  00 00        	.res	2,$00
000004r 1               _xhl:
000004r 1  00 00        	.res	2,$00

000006r 1               **.segment	"BSS"**
000000r 1               _z80:
000000r 1  00 00 00 00  	.res	32,$00
000004r 1  00 00 00 00  
000008r 1  00 00 00 00  

000020r 1               **.segment	"COMB"**
000006r 1               _globaladdressloc:
000006r 1  00 00        	.res	2,$00
000008r 1               _PC_VALUE:
000008r 1  00 00        	.res	2,$00
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

**Comment by mrdudz** [comment, marked as answer]
I can't tell why this happens.... but does the same happen when you do this:

```
#pragma bss-name (push, "COMB")
typedef struct
{
	unsigned int bc;       
	unsigned int de;       
	unsigned int hl;        
} DDREGS;
DDREG  ddregs;
```

? I'd expect this to work as you want it

**Reply by RetrofanC** [reply]
Thanks a million again for the quick reply!
Indeed that suggestion did the trick and all is working now again as expected. (-8
